### ISSUE TITLE
Solution explorer does not show Diagnostics nodes for analyzers added without canonical paths

### ISSUE THREAD

**Issue by pranavkm** [dotnet/roslyn#33920]
Sample app that reproduces the issue: https://github.com/pranavkm/analyzer-ruleset

Consider an analyzer added to the project as follows:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.2</TargetFramework>
  </PropertyGroup>
  
  <ItemGroup>
    <Analyzer Include="Microsoft.AspNetCore.Mvc.Api.Analyzers.dll"/>
  </ItemGroup>

</Project>
```

When inspected in Solution Explorer, the Analyzers node does not list diagnostics. E.g.

![image](https://user-images.githubusercontent.com/174281/53923280-98ee3f00-402c-11e9-91eb-b5967216252b.png)

If I change the path to be absolute, the diagnostics node appear correctly:

```xml
  <ItemGroup>
    <Analyzer Include="$(MSBuildThisFileDirectory)Microsoft.AspNetCore.Mvc.Api.Analyzers.dll"/>
  </ItemGroup>
```

@tmeschter's hypothesis is that the the language service knows the analyzer by it's canonical path which is different from the non-canonical path in the first case.

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback.
